splash/render_options.py

```diff
@@ -28,21 +35,23 @@ def fromrequest(cls, request, max_timeout):
 
         # 1. GET / POST data
         data = {key: values[0] for key, values in request.args.items()}
-        if request.method == 'POST':
-            content_type = request.getHeader('content-type')
+        if request.method == b'POST':
+            content_type = request.getHeader(b'content-type')
             if content_type:
                 request.content.seek(0)
 
                 # 2. application/json POST data
-                if 'application/json' in content_type:
+                if 'application/json' in content_type.decode('utf-8'):
                     try:
-                        data.update(json.load(request.content, encoding='utf8'))
+                        content = request.content.read()
+                        data.update(json.load(
+                            StringIO(content.decode('utf-8')), encoding='utf8'))
                     except ValueError as e:
-                        raise BadOption("Invalid JSON: '{}'".format(e.message))
+                        raise BadOption("Invalid JSON: '{}'".format(str(e)))
 
                 # 3. js_source from application/javascript POST requests
-                if 'application/javascript' in content_type:
-                    data['js_source'] = request.content.read()
+                if 'application/javascript' in content_type.decode('utf-8'):
```

Comment:
I like `b'application/javascript' in content_type` more because in all other places we use bytes constants for header names